### PR TITLE
Use type support handle functions from the rmw implementation

### PR DESCRIPTION
**Pull request by esteve**
Connects to ros2/ros2#18

### CODE REVIEW

**Comment by wjwwood** [comment]
I also just force pushed a rebase of `windows-disambiguate-linking` on top of `windows`.

**Comment by esteve** [comment]
@dirk-thomas @tfoote @wjwwood ready for review. I'll merge this when the other PRs in the ros2/ros2#10 card are ok'd.

**Comment by dirk-thomas** [comment]
lgtm

File: rclcpp/include/rclcpp/node_impl.hpp [hunk 1]
```diff
@@ -20,6 +20,9 @@
 #include <string>
 
 #include <rmw/rmw.h>
+#include <rmw/get_type_support_handle.h>
+#include <rmw/get_service_type_support_handle.h>
```
Comment (dirk-thomas) [inline rclcpp/include/rclcpp/node_impl.hpp]: This include seems to be not used?

**Comment by dirk-thomas** [comment]
Beside the comment: lgtm.

File: rclcpp/include/rclcpp/node_impl.hpp (hunk 1 above)
Comment (esteve) [inline rclcpp/include/rclcpp/node_impl.hpp]: They are used in lines 69 (https://github.com/ros2/rclcpp/blob/disambiguate-linking/rclcpp/include/rclcpp/node_impl.hpp#L69) and 100 (https://github.com/ros2/rclcpp/blob/disambiguate-linking/rclcpp/include/rclcpp/node_impl.hpp#L100)

Comment (dirk-thomas) [inline rclcpp/include/rclcpp/node_impl.hpp]: The first one yes, The second - service one - not yet.

**Comment by tfoote** [comment]
+1

**Pull request merged**